src/__tests__/__snapshots__/index.test.js.snap

```diff
@@ -33,6 +33,26 @@ Object {
 }
 `;
 
+exports[`denormalize denormalizes recursive dependencies 1`] = `
+Object {
```

Comment:
It's reasonable. I changed denormalization logic to confirm described expectations. Please, review last [commit](https://github.com/paularmstrong/normalizr/pull/220/commits/064ae64d4ca391a333204f4ddb3e8ab5a0ce3a71).